src/model/AirLoopHVACOutdoorAirSystem.cpp

```diff
@@ -612,6 +615,69 @@ namespace model {
       }
     }
 
+    ComponentType AirLoopHVACOutdoorAirSystem_Impl::componentType() const {
+      bool has_cooling = false;
+      bool has_heating = false;
+      for (const auto& comp : subsetCastVector<HVACComponent>(components())) {
+        auto compType = comp.componentType();
+        if (compType == ComponentType::Cooling) {
+          has_cooling = true;
+        } else if (compType == ComponentType::Heating) {
+          has_heating = true;
+        } else if (compType == ComponentType::Both) {
+          has_cooling = true;
+          has_heating = true;
+        }
+      }
+
+      // If source side is purely cooling
+      if (has_cooling && !has_heating) {
+        return ComponentType::Cooling;
+
+        // If source side is purely heating
+      } else if (!has_cooling && has_heating) {
+        return ComponentType::Heating;
+
+        // If there is nothing
+      } else if (!has_cooling && !has_heating) {
+        return ComponentType::None;
+
+        // All other cases: BOTH
+      } else {
+        return ComponentType::Both;
+      }
+    }
+
+    std::vector<FuelType> AirLoopHVACOutdoorAirSystem_Impl::coolingFuelTypes() const {
+      std::set<FuelType> result;
+      for (const auto& comp : subsetCastVector<HVACComponent>(components())) {
+        for (auto& ft : comp.coolingFuelTypes()) {
+          result.insert(ft);
+        }
+      }
+      return {result.begin(), result.end()};
+    }
```

Comment:
An AirLoopHVACOutdoorAirSystem is listed as a component on an AirLoopHVAC. So we need to implement the logic for AirLoopHVACOutdoorAirSystem as well.
Currently I'm checking each `components`, meaning regardless of whether it's on the **relief** stream or the **oa intake** stream. Does that sound right to you @asparke2

Reply:
Note to self: the HX:AirToAir:SensibleAndLatent appears **twice** in `components` (because it's on both stream). Even if we made it be something than no-op, that's fine, we're working with std::set